apps/authorization/models.py

```diff
@@ -70,11 +73,235 @@ def update_grants(*args, **kwargs):
 
 def check_grants():
     AccessToken = get_access_token_model()
-    token_count = AccessToken.objects.filter(
-        expires__gt=timezone.now(),
-    ).values('user', 'application').distinct().count()
+    token_count = (
+        AccessToken.objects.filter(
+            expires__gt=timezone.now(),
+        )
+        .values("user", "application")
+        .distinct()
+        .count()
+    )
     grant_count = DataAccessGrant.objects.all().count()
     return {
         "unique_tokens": token_count,
         "grants": grant_count,
     }
+
+
+def get_grant_counts():
+    # Grant real/synth bene counts (includes granted to multiple apps)
+    start_time = datetime.utcnow().timestamp()
+
+    real_bene_cnt = DataAccessGrant.objects.filter(
```

Comment:
would count on crosswalk directly give the same result?

    real_bene_distinct_cnt = (
        Crosswalk.objects.filter(~Q(fhir_id__startswith="-")).count()
    )

    synth_bene_distinct_cnt = (
        Crosswalk.objects.filter(Q(fhir_id__startswith="-")).count()
    )

no need to check fhir for null and empty, they are enforced by db constrain?

Reply:
@JFU-GIT  For grant and token aggregates with the Crosswalk, I was getting some slightly invalid counts in Sandbox until adding the "_isnull=False" to the queries.  I can't recall, but I think related to the Crosswalk default setting of "default=None".

From the constraints, there can only be one with FHIR_ID == "" so we want to skip that one, if it exists.